src/blsCompatibility.spec.ts

```diff
@@ -0,0 +1,1026 @@
+import { FpLegendre, isNegativeLE } from "@noble/curves/abstract/modular";
+import * as nobleUtils from "@noble/curves/abstract/utils";
+import { bls12_381 as nobleBls } from "@noble/curves/bls12-381";
+import { sha512 } from "@noble/hashes/sha512";
+import { assert } from "chai";
+import path from "path";
+import { readTestFile } from "./testutils/files";
+
+const Fp = nobleBls.fields.Fp;
+const Fp2 = nobleBls.fields.Fp2;
+const Fp12 = nobleBls.fields.Fp12;
+const G1 = nobleBls.G1;
+const G2 = nobleBls.G2;
+
+describe("test BLS compatibility (noble crypto and herumi)", () => {
+    before(() => {
+        setupG2GeneratorPointsLikeHerumi();
+    });
+
+    it("test using test vectors", async function () {
+        this.timeout(100000);
+
+        const numVectors = 256;
+        const testdataPath = path.resolve(__dirname, "testdata");
+        const filePath = path.resolve(testdataPath, "blsVectors.json");
```

Comment:
The JSON file will be moved to "mx-chain-crypto-go" at some point.

Reply:
A few of the "inline" test vectors (explicit, in tests code, not in the JSON file) were took from:

https://github.com/multiversx/mx-chain-crypto-go/pull/40